api/src/main/java/io/opencensus/metrics/export/Distribution.java

```diff
@@ -163,6 +130,103 @@ public static Distribution create(
    */
   public abstract List<Bucket> getBuckets();
 
+  /**
+   * The bucket boundaries used to create a histogram for the distribution.
+   *
+   * @since 0.17
+   */
+  @AutoValue
+  @Immutable
+  public abstract static class BucketOptions {
+
+    BucketOptions() {}
+
+    /**
+     * Creates a {@link BucketOptions}.
+     *
+     * @param explicitBuckets the explicit buckets with arbitrary widths.
+     * @return a {@code BucketOptions}.
+     * @since 0.17
+     */
+    public static BucketOptions create(Explicit explicitBuckets) {
+      Utils.checkNotNull(explicitBuckets, "explicitBuckets");
+      return new AutoValue_Distribution_BucketOptions(explicitBuckets);
+    }
+
+    /**
+     * Returns the {@link Explicit} associated with the {@link BucketOptions}.
+     *
+     * @return the {@code Explicit} associated with the {@code BucketOptions}.
+     * @since 0.17
+     */
+    public abstract Explicit getExplicitBuckets();
+  }
+
+  /**
+   * A Bucket with explicit bounds.
+   *
+   * @since 0.17
+   */
+  @Immutable
+  @AutoValue
+  public abstract static class Explicit {
+
+    Explicit() {}
+
+    /**
+     * Creates a {@link Explicit}.
+     *
+     * @param bucketBoundaries the bucket boundaries of a distribution (given explicitly).
+     * @return a {@code Explicit}.
+     * @since 0.17
+     */
+    public static Explicit create(List<Double> bucketBoundaries) {
+      Utils.checkNotNull(bucketBoundaries, "bucketBoundaries list should not be null.");
+      return new AutoValue_Distribution_Explicit(copyBucketBounds(bucketBoundaries));
+    }
+
+    private static List<Double> copyBucketBounds(List<Double> bucketBoundaries) {
+      Utils.checkNotNull(bucketBoundaries, "bucketBoundaries list should not be null.");
+      List<Double> bucketBoundariesCopy = new ArrayList<Double>(bucketBoundaries); // Deep copy.
+      // Check if sorted.
+      if (bucketBoundariesCopy.size() > 1) {
```

Comment:
You'll also need to check if the bucket bounds are positive.

Reply:
Added check for first element. Rest will be covered in sorted logic.